### PR TITLE
Release v0.5

### LINKED ISSUE #201: v0.5.0 release announcement and docs
- release blogpost
- set v0.5.0 as the default docs version (rename v0.5.0-snapshot)
- add new v0.6.x-snapshot docs version

### PR DESCRIPTION
**Pull request by jldec**
Resolves #201 

- [x] release blogpost
- [x] set v0.5.0 as the default / stable docs version
- [x] add new v0.6.x-snapshot docs version
- [x] fix getting started and streaming runtime pages to point to final release artifacts

[Preview blogpost](https://deploy-preview-202--projectriff.netlify.com/blog/2020/02/07/announcing-riff-0-5-0)

### CODE REVIEW

**Review by scothis — DISMISSED** [review]
/hold until everything else is ready to announce

**Review by trisberg — CHANGES_REQUESTED** [review]
The getting started docs still have `0.5.0-snapshot` links for the yaml files.

And also runtimes docs for streaming and core.

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]:
```suggestion
We are happy to announce riff v0.5.0. Thanks to all riff, Cloud Native Buildpack, Knative, Liiklus, KEDA, Contour, and other contributors.
```

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]: if you want to be more fancy:

```js
module.exports = ({ impressions, orders }, { conversions }) => {
     // do something
 };
 module.exports.$interactionModel = 'node-streams';
```

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]: this isn't new, it existed in 0.4

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]: we should use a diagram that explains how 0.5 is implemented, not what we're thinking about for 0.6

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]:
```suggestion
- We added Ingress support to the Core Runtime, allowing external HTTP requests to target functions, apps, and containers.
```

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]: things worth calling out that are not discussed:

- kapp is encouraged for installs
- helm is no longer supported for installs
- each runtime (core, knative, streaming) can be installed independently at any time, and uninstalled independently
- Deployers and Processors now consistently use PodTemplateSpec to define containers and volumes
- CRD status now favor object references instead of resource names

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]:
```suggestion
[Gateways](docs/v0.5/cli/riff-streaming-gateway) connect streams to a messaging service like Kafka or Pulsar. [Processors](/docs/v0.5/cli/riff-streaming-processor-create) connect functions with gateways.
```

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]:
```suggestion
This release includes Gateways for Kafka and Pulsar, and a non-production in-memory gateway. All three are based on [Liiklus](https://github.com/bsideup/liiklus).
```

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]: The discussions are broader than the knative community
```suggestion
We are discussing this streaming model with the community, and exploring ways to enable streaming for other types of workloads beyond streaming functions.
```

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]: should we call out that we're considering deprecating the core runtime?

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]:
```suggestion
- [Contour](https://projectcontour.io) has replaced Istio as the ingress controller, and works with both the Knative and the Core runtime. Note that Deployers are only accessible from inside the cluster by default, `--ingress-policy External` is now [required](/docs/v0.5/cli/riff-knative-deployer-create) in order to route HTTP traffic from outside the cluster.
```

File: docs/v0.6/versions/v0.6.md
Comment (scothis) [inline docs/v0.6/versions/v0.6.md:37]: can we use netlify redirects instead of these static hacks?

**Comment by jldec** [comment]
@trisberg @scothis @ekcasey 
I won't be able to fix this tonight, but please go ahead and push the changes necessary.
thx

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (scothis) [inline website/blog/2020-02-07-announcing-riff-0-5-0.md]: we should talk a bit about how this is different than what riff has had in the past

File: docs/v0.6/versions/v0.6.md
Comment (jldec) [reply docs/v0.6/versions/v0.6.md:37]: maybe next time (it introduces lock-in)

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (jldec) [reply website/blog/2020-02-07-announcing-riff-0-5-0.md]: This was based on https://github.com/projectriff/system/commit/c69abe9d3c9c526b744d77f1392ba8b6b533d911.

I'll remove the section for now, or someone can try to reword.
cc. @ekcasey

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (jldec) [reply website/blog/2020-02-07-announcing-riff-0-5-0.md]: I'd prefer to make a more general statement once we have a clearer picture of the riff + Knative plan.

**Review by trisberg — COMMENTED** [review]
Removed -snapshot from 0.5.0

**Review by trisberg — APPROVED** [review]

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (jldec) [reply website/blog/2020-02-07-announcing-riff-0-5-0.md]: That's cool - i think we should stick with the slightly longer version for clarity.

File: website/blog/2020-02-07-announcing-riff-0-5-0.md
Comment (jldec) [reply website/blog/2020-02-07-announcing-riff-0-5-0.md]: The text mentions multiple input streams and gateways. I'd rather leave it like that while we're still evolving rather than drawing comparisons with specific designs in the past.

**Comment by scothis** [comment]
The cli release has been published, homebrew and chocolatey packages will need to be updated in homebrew and chocolatey. Homebrew took a few days to get approval last time. Let's not wait for it.

**Pull request merged**
